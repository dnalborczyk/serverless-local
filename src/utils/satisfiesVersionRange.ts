import { satisfies, valid, validRange } from 'semver'

export default function satisfiesVersionRange(
  version: string,
  range: string,
): boolean {
  if (valid(version) == null) {
    throw new Error(`Not a valid semver version: ${version}`)
  }

  if (validRange(range) == null) {
    throw new Error(`Not a valid semver range: ${range}`)
  }

  return satisfies(version, range)
}
