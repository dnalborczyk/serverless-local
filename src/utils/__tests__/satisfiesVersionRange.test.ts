import satisfiesVersionRange from '../satisfiesVersionRange'

describe('satisfiesVersionRange', () => {
  describe('valid parameters', () => {
    test.each`
      version      | range       | expected | description
      ${'1.38.0'}  | ${'>=1.38'} | ${true}  | ${'same as minimum'}
      ${'1.40.0'}  | ${'>=1.38'} | ${true}  | ${'greather than minimum'}
      ${'1.37.11'} | ${'>=1.38'} | ${false} | ${'less than minimum'}
    `(
      'should return $expected when $description is passed',
      ({ expected, range, version }) => {
        const result = satisfiesVersionRange(version, range)
        expect(result).toEqual(expected)
      },
    )
  })

  describe('should throw when invalid parameters are passed', () => {
    test.each`
      version     | range       | description
      ${'a.b.c'}  | ${'>=1.40'} | ${'invalid version'}
      ${'1.40.0'} | ${'a.b.c'}  | ${'invalid range'}
    `(
      'should throw error when $description is passed',
      ({ range, version }) => {
        expect(() =>
          satisfiesVersionRange(version, range),
        ).toThrowErrorMatchingSnapshot()
      },
    )
  })
})
