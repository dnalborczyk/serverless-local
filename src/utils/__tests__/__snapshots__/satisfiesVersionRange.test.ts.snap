// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`satisfiesVersionRange should throw when invalid parameters are passed should throw error when invalid range is passed 1`] = `"Not a valid semver range: a.b.c"`;

exports[`satisfiesVersionRange should throw when invalid parameters are passed should throw error when invalid version is passed 1`] = `"Not a valid semver version: a.b.c"`;
